mito-ai/src/utils/notebook.tsx

```diff
@@ -339,4 +340,122 @@ export const scrollToCell = (
 
     // Use the new JupyterLab scrollToCell method instead of DOM node scrollIntoView
     void notebookPanel.content.scrollToCell(cell, position);
+}
+
+export const scrollToNextCellWithDiff = (
+    notebookTracker: INotebookTracker,
+    currentCellId: string,
+    changedCells: { cellId: string, originalCode: string, currentCode: string }[],
+): void => {
+    // Early return if no diffs remain
+    if (changedCells.length === 0) {
+        return;
+    }
+
+    // Convert changedCells to agentEdits format for internal use
+    const agentEdits = changedCells.map(change => ({
+        cellId: change.cellId,
+        code: change.currentCode
+    }));
+
+    // Find current cell's position in the edits list
+    const currentEditIndex = agentEdits.findIndex(edit => edit.cellId === currentCellId);
+    if (currentEditIndex < 0) {
+        return; // Current cell not found in edits
+    }
+
+    // Find the next cell that still has a diff
+    const nextCellWithDiff = agentEdits
+        .slice(currentEditIndex + 1)
+        .find(edit => changedCells.some(change => change.cellId === edit.cellId));
+
```

Comment:
1. If you have the agent create 3 cells
2. Click Review
3. Instead of resolving the first cell, you click on the last of the 3 new cells
4. Click Accept
5. This code doesn't identify that it should now take you back to the first cell. 

I think to implement this, we can:
1.  Remove the cells from the `changedCellsRef` as we Accept/Reject them because that is what we are feeding to this function as the `changedCells` it looks like. 
2. When we reach the end of the list, if there are still cells left in `changedCells`, we go back to the start of the list

https://github.com/user-attachments/assets/44d11d4e-8f9c-4d95-9613-2e3fba0995ac